src/algebra/lattice_ordered_group.lean

```diff
@@ -0,0 +1,595 @@
+is said to be a *lattice ordered (additive) group*.
+-/
+class lattice_ordered_add_group (α : Type u)
+  extends add_group α, lattice α :=
+(add_le_add : ∀ a b : α, a ≤ b → ∀ c d : α, c + a + d ≤ c + b + d)
+
+/--
+A lattice ordered additive group with Abelian group addition is said to be a *lattice ordered
+(additative) commutative group*.
+-/
+class lattice_ordered_add_comm_group (α : Type u)
+  extends add_comm_group α, lattice α :=
+(add_le_add_left : ∀ a b : α, a ≤ b → ∀ c : α, c + a ≤ c + b)
+
+-- Tells to_additive that lattice_ordered_add_group is the additive form of lattice_ordered_group
+-- and lattice_ordered_add_comm_group is the additive form of lattice_ordered_comm_group
+attribute [to_additive] lattice_ordered_group
+attribute [to_additive] lattice_ordered_comm_group
+
+-- Every lattice ordered commutative group is also an ordered additive commutative group
+@[priority 100, to_additive] -- see Note [lower instance priority]
+instance lattice_ordered_comm_group.to_ordered_comm_group (α : Type u)
+  [s : lattice_ordered_comm_group α] : ordered_comm_group α :=
+{ mul := s.mul, ..s }
+
+-- A lattice ordered commutative group is a lattice ordered group
+@[priority 100, to_additive] -- see Note [lower instance priority]
+instance lattice_ordered_comm_group.to_lattice_ordered_group (α : Type u)
+  [s : lattice_ordered_comm_group α] : lattice_ordered_group α :=
+{ mul_le_mul := λ a b hab c d, mul_le_mul' (mul_le_mul' le_rfl hab) le_rfl }
+
+
+-- A linearly ordered additive commutative group is a lattice ordered commutative group
+@[priority 100, to_additive] -- see Note [lower instance priority]
+instance linear_ordered_comm_group.to_lattice_ordered_comm_group (α : Type u)
+ [s : linear_ordered_comm_group α] : lattice_ordered_comm_group α :=
+{ mul_le_mul_left := s.mul_le_mul_left, }
+
+variables {α : Type u} [lattice_ordered_add_comm_group α]
+
+-- Special case of Bourbaki A.VI.9 (1)
+/--
+Let `α` be a lattice ordered commutative group. For all elements `a`, `b` and `c` in `α`,
+$$c + (a⊔b) = (c+a)⊔(c+b).$$
+-/
+lemma add_sup_eq_add_sup_add (a b c : α) : c + (a⊔b) = (c+a)⊔(c+b) :=
+begin
+  refine le_antisymm _ (by simp),
+  rw [← add_le_add_iff_left (-c), ← add_assoc, neg_add_self, zero_add],
+  apply sup_le,
+  { simp },
+  { simp },
+end
+
+-- Special case of Bourbaki A.VI.9 (2)
+/--
+Let `α` be a lattice ordered commutative group. For all elements `a` and `b` in `α`,
+$$-(a⊔b)=(-a)⊓(-b).$$
+-/
+lemma neg_sup_eq_neg_inf_neg (a b : α) : -(a⊔b)=(-a)⊓(-b) :=
+begin
+  rw le_antisymm_iff,
+  split,
+  { rw le_inf_iff,
+    split,
+    { rw neg_le_neg_iff, apply le_sup_left, },
+    { rw neg_le_neg_iff, apply le_sup_right, } },
+  { rw ← neg_le_neg_iff, simp,
+    split,
+    { rw ← neg_le_neg_iff, simp, },
+    { rw ← neg_le_neg_iff, simp, } }
+end
+
+/--
+Let `α` be a lattice ordered commutative group. For all elements `a` and `b` in `α`,
+$$-a⊔-b = -(a⊓b).$$
+-/
+@[simp] lemma sup_neg_eq_neg_inf (a b : α) : -a⊔-b = -(a⊓b) :=
```

Comment:
I think you should flip this lemma to match `neg_sup_eq_neg_inf_neg` or vice-versa.

Reply:
According to https://leanprover-community.github.io/extras/simp.html#simp-lemmas the expression on the right hand side should be simpler than that on the left hand side. So that leaves me trying to decide if `-a⊔-b` is simpler than `-(a⊓b)`? By comparison with `mul_inv_rev (a b : G) : (a * b)⁻¹ = b⁻¹ * a⁻¹` it would appear that `-a⊔-b` is the simpler form?

Reply:
Done, now called `inv_inf_eq_sup_inv`.